src/api/routes/adminRouter.js

```diff
@@ -0,0 +1,57 @@
+import { Router } from 'express';
+import authController from '../controllers/authController';
+import adminPermissionsController from '../controllers/adminPermissionsController';
+import ValidateUser from '../../middlewares/ValidateUser';
+import validateToken from '../../middlewares/checkValidToken';
+import checkUser from '../../middlewares/checkUser';
+import checkIfAdmin from '../../middlewares/IsAdmin';
+import checkIfBlocked from '../../middlewares/isBlocked';
+
+const adminRouter = new Router();
+
+adminRouter.post('/users/create',
+  validateToken,
+  checkIfAdmin.checkAdmin,
+  ValidateUser.validateSignup,
+  authController.signup);
+
+adminRouter.get('/users/fetch',
```

Comment:
@Ramadhan0 instead of `users/fetch`, add `admin` to all admin routes. E.g:

`/admin/users`, 
`/admin/users/:email`
`/admin/users/moderator/:email`

No need to have `create` or `update` in your route. The HTTP verb determines what controller method handles the request.

Please, apply this feedback across all your routes.